Documentation update: How To Guides overview (#367)
aragon/osx: packages/contracts/docs/osx/02-how-to-guides/01-dao/01-best-practices.md

title: Best Practices
---

## Some Advice When Operating your DAO

### DOs 👌

> - Make sure that at least one address (typically a governance plugin) has `EXECUTE_PERMISSION_ID` permission so that something can be execute on behalf of the DAO.

Reviewer: ```suggestion
- Make sure that at least one address (typically a governance plugin) has `EXECUTE_PERMISSION_ID` permission so that something can be executed on behalf of the DAO.
```

> - Check every proposal asking to install, update, or uninstall a plugin with utmost care and review. Installation means granting an external contract permissions, so you want to be extra careful about:

Reviewer 2: It is not only about contracts. Addresses in general get permissions and these can also be accounts.
It is also not only about "Installation" but for every setup type (Installation, Update, Uninstallation) there must be a very careful review of the items below.

I would leave the sentence as it was

  - the implementation contract
  - the setup contract
  - the helper contracts
  - the permissions

### DON'Ts ✋

> - Incapacitate your DAO by revoking the last `EXECUTE_PERMISSION_ID`. This means your DAO will be blocked and any assets you hold may be locked in forever. Revoking `EXECUTION_PERMISSION_ID` is done through:

Reviewer 2: ```suggestion
- Incapacitate your DAO by revoking the last `EXECUTE_PERMISSION_ID`. This means your DAO will be blocked and any assets you hold may be locked in forever. This can happen through:
```

  - uninstalling your last governance plugin.
  - applying an update to your last governance plugin.

> - You shouldn't create a DAO with no plugins, or with plugins that can’t effectively execute. This will block your DAO from ever being used, since you need some kind of governance mechanism to execute actions.

Reviewer 2: "since you need some kind of governance mechanism to execute actions."
This statement is not true. A caller (e.g., a wallet) can directly execute actions without a governance mechanism if it has the `EXECUTE_PERMISSION` granted to it.

I would leave this part out.

Author: Nice! Will edit.
